### PR TITLE
Add recording and alerting rules for TrustyAI

### PR DESCRIPTION
**Pull request by ruivieira**
Add recording and alerting rules for TrustyAI ([RHOAIENG-99](https://issues.redhat.com/browse/RHOAIENG-99)).

Note: SOPs not merged yet.

### CODE REVIEW

File: config/monitoring/prometheus/apps/prometheus-configs.yaml [hunk 1]
```diff
@@ -1396,3 +1415,87 @@ data:
...
+            sum(probe_success:burnrate30m{instance=~"trustyai-service-operator-controller-manager"}) by (instance) > (6.00 * (1-0.98000))
+            and
+            sum(probe_success:burnrate6h{instance=~"trustyai-service-operator-controller-manager"}) by (instance) > (6.00 * (1-0.98000))
+          for: 15m
+          labels:
+            severity: critical
+        - alert: TrustyAI Controller Probe Success Burn Rate
+          annotations:
+            message: 'High error budget burn for {{ $labels.instance }} (current value: {{ $value }}).'
+            triage: "https://gitlab.cee.redhat.com/service/managed-tenants-sops/-/blob/main/RHODS/Model-Serving/rhods-trustyai-controller-probe-success-burn-rate.md"
```
Comment (zdtsw) [inline config/monitoring/prometheus/apps/prometheus-configs.yaml:1493]: dont think this is a valid link.
have you added new SLO doc in the gitlab repo?

File: config/monitoring/prometheus/apps/prometheus-configs.yaml [hunk 2]
```diff
@@ -1396,3 +1415,87 @@ data:
...
+            1 - min(avg_over_time(probe_success{instance="trustyai-service-operator-controller-manager"}[5m]))
+          labels:
+            instance: trustyai-service-operator-controller-manager
+          record: probe_success:burnrate5m
+        - expr: |
+            1 - min(avg_over_time(probe_success{instance="trustyai-service-operator-controller-manager"}[6h]))
+          labels:
+            instance: trustyai-service-operator-controller-manager
+          record: probe_success:burnrate6h
+    trustyai-alerting.rules: |
```
Comment (zdtsw) [inline config/monitoring/prometheus/apps/prometheus-configs.yaml:1462]:
```suggestion
  trustyai-alerting.rules: |
```
indentation

File: config/monitoring/prometheus/apps/prometheus-configs.yaml (hunk 1 above)
Comment (ruivieira) [reply config/monitoring/prometheus/apps/prometheus-configs.yaml:1493]: Hi @zdtsw, SLOs not merged yet, will finish them ASAP and ping here.

File: config/monitoring/prometheus/apps/prometheus-configs.yaml (hunk 2 above)
Comment (zdtsw) [reply config/monitoring/prometheus/apps/prometheus-configs.yaml:1462]: will you check this part? @ruivieira

**Review by zdtsw — APPROVED** [review]
/lgtm

**Comment by zdtsw** [comment]
@asanzgom @mattmahoneyrh this will need verification in 2.6 as well.

**Pull request merged**